ouroboros-network-api/src/Ouroboros/Network/PeerSelection/RelayAccessPoint.hs

```diff
@@ -53,6 +57,35 @@ data RelayAccessPoint = RelayAccessDomain  !DNS.Domain !Socket.PortNumber
                       | RelayAccessAddress !IP.IP      !Socket.PortNumber
   deriving (Eq, Ord)
 
+-- | These instances are used to serialize 'LedgerPeerSnapshot'
+-- consensus LocalStateQuery server which uses these instances
+-- for all its query responses. It appears they provide some improved
+-- debugging diagnostics over Serialize instances.
+instance ToCBOR RelayAccessPoint where
+  toCBOR = mconcat . \case
+    RelayAccessDomain domain port ->
+      [encodeWord8 0, serialize' port, toCBOR domain]
+    RelayAccessAddress (IP.IPv4 ip4) port ->
+      [encodeWord8 1, serialize' port, toCBOR (IP.fromIPv4 ip4)]
+    RelayAccessAddress (IP.IPv6 ip6) port ->
+      [encodeWord8 2, serialize' port, toCBOR (IP.fromIPv6 ip6)]
+    where
+      serialize' = toCBOR . toInteger
+
+instance FromCBOR RelayAccessPoint where
+  fromCBOR = do
+    mode <- decodeWord8
+    port <- fromInteger <$> fromCBOR @Integer
+    if | mode == 0 ->  do
+         domain <- fromCBOR
+         return $ RelayAccessDomain domain port
+       | mode == 1 -> do
+         ip4 <- IP.IPv4 . IP.toIPv4 <$> fromCBOR
+         return $ RelayAccessAddress ip4 port
+       | mode == 2 -> do
+         ip6 <- IP.IPv6 . IP.toIPv6 <$> fromCBOR
+         return $ RelayAccessAddress ip6 port
+
```

Comment:
This PR should also include a test that `fromCBOR` and `toCBOR` are compatible, e.g.
```hs
 fromCBOR . toCBOR = id
```

But also check that the `toCBOR` produces valid CBOR.  This is especially important if we don't have CDDL spec for this definition.   This can be done by encoding and then decoding with [`decodeTermToken`](https://hackage.haskell.org/package/cborg-0.2.10.0/docs/Codec-CBOR-FlatTerm.html#v:decodeTermToken) and validating with ` [`validFlatTerm`](https://hackage.haskell.org/package/cborg-0.2.10.0/docs/Codec-CBOR-FlatTerm.html#v:validFlatTerm); check `prop_codec_valid_cbor_encoding`.

Reply:
Done.